SubscribeIT/ngDesk
# 266 company signup improvement

#### Reference to issue (Required)

Closes #266 

#### Checklist (Required)


#### Description (Required)

Company signup improvement

#### Manual Test Preformed (Required)
List the manual tests that were preformed and the steps to preform them  
**There must be multiple test cases provided, each test case should include the exact and detailed steps to replicate the case**
**I have included 3, test cases here but there will be many time when more cases are required**


##### Test Case 1
**Name: POST FEW NEW COMPANY  WITH DIFFERENT SUBDOMAIN**

- go to postman.
- given post call URL:[https://dev1.ngdesk.com/api/ngdesk-company-service-v1/company].
- added proper payload for post call for posting new company
- perform post
- able to post new company without any errors
- able to login in ngdesk with valid credential

##### Test Case 2
**Name:TEST FOR DAILY COMPANY SIGNUP**

- go to company service in eclipse
- go to  package com.ngdesk.company.dao -> CompanySignUpJobservice
- in sendEmail put log for body
- go to DailyCompanySignupJob in package com.ngdesk.company.jobs;
- give @Scheduled(fixedRate = 6000) and comment cron Schedule
- comment the return of environment check
- able to see logs for body in console
- go to https://www.w3schools.com/tryit/tryit.asp?filename=tryhtml_hello
- copy the logs and paste in html editor and able to see all company details

![Screenshot from 2021-11-02 10-25-06](https://user-images.githubusercontent.com/89505098/139790670-6be0e9bc-5787-47e4-b2b4-44a1a87f1cda.png)

![image](https://user-images.githubusercontent.com/89505098/139790774-462d6912-e8f6-488f-a800-f0a773075c77.png)
![image](https://user-images.githubusercontent.com/89505098/139791214-25365a11-0fd6-4bba-9094-584ed9020ad5.png)

##### Test Case 3
**Name:TEST FOR MONTHLY COMPANY SIGNUP**

- go mongodb -> companies collection
- in few company  modified date created field (give previous month date)
- go to company service in eclipse
- go to  package com.ngdesk.company.dao -> CompanySignUpJobservice
- in sendEmail put log for body
- go to MonthlyCompanySignupJob in package com.ngdesk.company.jobs;
- give @Scheduled(fixedRate = 6000) and comment cron Schedule
- comment the return of environment check
- able to see logs for body in console
- go to https://www.w3schools.com/tryit/tryit.asp?filename=tryhtml_hello
- copy the logs and paste in html editor and able to see all company details

![Screenshot from 2021-11-02 10-24-58](https://user-images.githubusercontent.com/89505098/139791158-c45fc8b9-719d-4a59-b192-2fef9bb5aa1f.png)
![Screenshot from 2021-11-02 10-25-06](https://user-images.githubusercontent.com/89505098/139791171-15185833-b5cd-4b94-bd82-36b00c74818f.png)
![image](https://user-images.githubusercontent.com/89505098/139791275-d15457fd-f9a9-4c3e-a375-e5e199e3f6a6.png)


#### List of General Components affected (Required)

**COMPANY SERVICE AND REST**

#### Types of changes (Required)
